Add `to_timeseries` and `from_timeseries` to support AstroPy TimeSeries objects

barentsen opened the pull request:
This PR adds the `LightCurve.to_timseries()` and `LightCurve.from_timeseries()` methods to work with the prototype `TimeSeries` objects which are being developed as part of AstroPy ([link](https://astropy-timeseries.readthedocs.io)).

Example:
```Python
import lightkurve as lk
lc = lk.LightCurve(time=[50001, 50002, 50003], flux=[1, 2, 3], time_format='mjd')
lc.to_timeseries()
```

For a demo, see this Notebook gist: https://gist.github.com/barentsen/1a938ede20bd26ce3dca74637d8b1aee

/cc @astrofrog @gully

astrofrog commented:
Looks great!

gully commented:
Yeah, looks great!  One thing it can't do right now is round-trip metadata.  For example, the `quality ` column exists going from `LightCurve` to `TimeSeries` but then is lost going from `TimeSeries` to `LightCurve`.

barentsen commented:
I rebased on master, and added a clear error message in case someone tries to use `to_timeseries` without having AstroPy>=3.2 installed.  I intend to merge this shortly so we can start using `TimeSeries` in earnest!

barentsen commented:
(As a bonus, I also added a unit test!)

barentsen commented:
Awesome. Merging!
